dateutil/rrule.py

```diff
@@ -447,6 +448,20 @@ def __init__(self, freq, dtstart=None,
             until = datetime.datetime.fromordinal(until.toordinal())
         self._until = until
 
+        if self._dtstart and self._until:
```

Comment:
There's no need for this line. The problem is with the incompatibility between DTSTART and UNTIL. Not sure what the RFC has to say, but if UNTIL is aware and DTSTART has no specified time zone, you still don't know when to end the rule.

Reply:
It seems necessary to me.  If I change this and run the tests:

```python
>       if (self._dtstart.tzinfo is not None) != (self._until.tzinfo is not None):
E       AttributeError: 'NoneType' object has no attribute 'tzinfo'
```

Reply:
...or at the very least, we need to check truthiness of `self._until`.

Reply:
Ah, you are right, sorry, I read this as `self._dstart.tzinfo`, my bad.